src/main/java/Entities/ProductComparatorController.java

```diff
@@ -0,0 +1,30 @@
+package Entities;
+
+/**
+ * a controller class that uses comparator factory to create the different types of comparators used in sorting
+ */
+public class ProductComparatorController {
+    /**
+     * Returns a ProductComparator based off word
+     * @param word String value that corresponds to what type of comparator is created
+     * @return ProductComparator that is created in the method depending on word
+     */
+    public ProductComparator sortWishlist (String word) {
+        ProductComparatorFactory ProductFactory = new ProductComparatorFactory();
+
+        switch (word.toLowerCase()) {
+            case "date":
```

Comment:
This switch case can be simplied to return productFactory.createComparator(word.toLowerCase());